pack/nodejs/helm.ts

```diff
@@ -0,0 +1,715 @@
+import { execSync } from "child_process";
+import * as fs from "fs";
+
+import * as k8s from "./index";
+import * as pulumi from "@pulumi/pulumi";
+import * as shell from "shell-quote";
+import * as tmp from "tmp";
+import * as yaml from "js-yaml";
+
+export namespace v2 {
+    export interface ChartOpts {
+        repo: string;
+        chart: string;
+        version: string;
+
+        values?: any;
+        transformations?: transform.Transform[];
+        fetchOpts?: FetchOpts;
+    }
+
+    // A CustomResource representing a Helm Chart.
+    //
+    // `Chart` does not use Tiller. The Chart specified is copied and expanded locally; any values
+    // that would be retrieved in-cluster would be assigned fake values, and none of Tiller's
+    // server-side validity testing is executed.
+    //
+    // Important caveat: Charts often assign hard-coded names for objects; since Pulumi is not able
+    // to assign such names, any time a resource must be replaced, it will be marked as
+    // delete-before-replace. For example, in a normal Pulumi program, a change in a ConfigMap's
+    // would result in: (1) a new ConfigMap being created, (2) all Deployments referencing that
+    // ConfigMap being updated to refer to the new ConfigMap, and (3) deleting the old ConfigMap
+    // only after all the rollouts of those Deployments have succeeded. In contrast, since the
+    // chart asks for a specific name for this ConfigMap, it must be deleted before being replaced
+    // with a ConfigMap of the same name.
```

Comment:
I'd focus more on "preserving the same semantics as Helm" here and simplify the comment.

Reply:
But the issue is that the `update` semantics _don't_ match Helm's. Helm will update the configmap in place, but _not_ trigger a rolling update of deployments.

Reply:
Oh actually, you're right. These are the same semantics as helm. We delete and then replace.